features/step_definitions/user_steps.rb

```diff
@@ -151,26 +158,17 @@ def page_heading_text(page)
   click_on('Register')
 end
 
-Given('they login as {string}') do |username|
-  fill_in('username', with: username)
-  fill_in('password', with: 'bar')
-  click_on('SignIn')
-  click_on('I Agree')
-end
+Given(/^they login as "(.*)"( with a random pid)?$/) do |user_string, with_random_pid|
+  if user_string == "the newly registered user"
+    user_string = @username
+  end
 
-Given('they login as the newly registered user') do
-  fill_in('username', with: @username)
-  fill_in('password', with: 'bar')
-  click_on('SignIn')
-  click_on('I Agree')
-end
+  log_in_as(user_string)
+
+  if with_random_pid
```

Comment:
I meant you don't need the outer `if` statement at all. Just this line is enough: `page.execute_script('document.getElementById("randomPid").value = "true"') if with_random_pid`
It has the if check appended at the end, which all we need.

```suggestion
```

Reply:
https://github.com/alphagov/verify-acceptance-tests/blob/c5942d86b5b0c7b412701f5d69fb140682b3deb5/features/step_definitions/user_steps.rb#L168-L170

Reply:
@CharlesIC Got it. Change made. Please re-review. Thanks